src/json_printer.sql

```diff
@@ -55,40 +70,65 @@ package body "JSON_PRINTER" as
       return str;
     end if;
   end llcheck;  
-
-  function escapeString(str varchar2) return varchar2 as
-    sb varchar2(32767) := '';
-    buf varchar2(40);
-    num number;
+  
+  -- escapes a single character. 
+  function escapeChar(ch char) return varchar2 deterministic is
```

Comment:
I'd like to see this function use a single return statement. There's no short circuiting being done in this function, so it doesn't help to have an early return statement.